src/registry/GithubRegistry.ts

```diff
@@ -167,6 +170,11 @@ export class GithubRegistry extends BaseRegistry implements IRegistry {
     return `https://raw.githubusercontent.com/${this.repoOwner}/${this.repoName}/${this.branch}/${path}`;
   }
 
+  protected getApiUrl(): string {
+    const githubApiUrl = `https://api.github.com/repos/${this.repoOwner}/${this.repoName}/git/trees/${this.branch}?recursive=true`;
+    return this.proxyUrl ? `${this.proxyUrl}/${githubApiUrl}` : githubApiUrl;
```

Comment:
So the url for a proxy would be like
`https://proxy.hyperlane.xyz/https://api.github.com/...` ? 
That looks odd to me. Does that work? I'd expect us to need to URL encode the path at the very least.

Reply:
Updated to have without the entire path ie. https://proxy.hyperlane.xyz/repos/hyperlane-xyz/hyperlane-registry/git/trees/main?recursive=true